### PR TITLE
various bugfixes

### PR DESCRIPTION
**Pull request by winglian**
use latest tinyllama release
check if val_set_size is empty first
update sdp and xformers llama patches for updated upstream transformers
fix system prompt when no input
calculate total and total supervised tokens even when not sample packing

### CODE REVIEW

**Comment by NanoCode012** [comment]
Hey, I recall our conversation about some ValueError: Expected len>=0. Was that issue fixed? If not, can it be combined here.

**Comment by winglian** [comment]
> Hey, I recall our conversation about some ValueError: Expected len>=0. Was that issue fixed? If not, can it be combined here.

just pushed a fix for that too. thanks!

File: src/axolotl/monkeypatch/llama_attn_hijack_sdp.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def sdp_attention_forward(
     past_key_value: Optional[Tuple[torch.Tensor]] = None,
     output_attentions: bool = False,
     use_cache: bool = False,
+    padding_mask: Optional[torch.LongTensor] = None,  # pylint: disable=unused-argument
```
Comment (NanoCode012) [inline src/axolotl/monkeypatch/llama_attn_hijack_sdp.py:28]: Should you add kwarg to prevent future break on update?

**Review by NanoCode012 — APPROVED** [review]
Not sure if the kwargs is also needed on every other patch, but this looks good to me!

**Pull request merged**